elfo-core/src/message.rs

```diff
@@ -150,15 +150,21 @@ impl Serialize for AnyMessage {
     where
         S: serde::ser::Serializer,
     {
-        let mut tuple = serializer.serialize_tuple(3)?;
-        tuple.serialize_element(self.protocol())?;
-        tuple.serialize_element(self.name())?;
-
         // TODO: avoid allocation here
         let erased_msg = self._erase();
-        tuple.serialize_element(&*erased_msg)?;
-
-        tuple.end()
+        if crate::scope::serde_mode() == SerdeMode::Dumping {
```

Comment:
We should comment that `Serialize` and `Deserialize` instances run only if `AnyMessage` is used inside other messages (although it's unstable anyway).